packages/devtools-launchpad/src/server/launch.js

```diff
@@ -23,6 +24,13 @@ function handleLaunchRequest(req, res) {
         tcpPort: getValue("firefox.tcpPort")
       };
       if (!isRunning) {
+        process.on('unhandledRejection', (err)=> {
+          if (err.message.indexOf('Could not locate Firefox on the current system')>-1) {
```

Comment:
I have a feeling that depending on the users localization this could only work on English based systems. Is there any way to do a test that isn't hard coded to the text of the message?

Reply:
I don't know better way to do this, if it possible i would rather have the promise rejected instead of catching the error thrown on the process. I already mention that.

Reply:
Then let's look at this problem slightly differently. Is the error message really that important to rework? Instead of analyzing the message in the first place, could we instead check for error and if found display out own little fail message followed by a newline and then the original error message itself?

By not doing any checking or manipulation the error from the underlying program gets out to the user. That could end up being more useful in different error states as well rather than this one specific case being looked for. It also avoids any internationalization issues. Overall, making the code simpler, more robust, and better for end-users in more situations.

Reply:
I cant argue with your point, but also i won't digging on selenium driver. Also the probability user install `custom location` is really low. So i leaving the firefox case as is.